File: Lab1/1001761950_calculator.py
# create the functions to add two numbers and return sum
def add(num1, num2):
    return num1 + num2

# create the functions to subtract number 2 from number 1 and return difference
def subtract(num1, num2):
    return num1 - num2

# create the functions to multiply two numbers and return sum
def multiply(num1, num2):
    return num1 * num2

# create the functions to divide number 1 by number 2 and return result
def divide(num1, num2):
    #make sure y is not 0 b/c that will cause an error
    if num2 == 0.0:
        return "You can't divide by 0 in this universe!"
    else:
        return num1 / num2


def calculator():
    #create the number variables and operators to
    #make sure the computer know it is the right daata type (float and String)
    num1 = 0.0
    num2 = 0.0
    operator = ""
    answer = 0.0
    inStr = ""

    # keep asking user until either user quits or 
    while True:
        # ask the user for inputs (num1, num2)
        inStr = input("Enter expression (or Q to quit): ")
        if inStr == 'q' or inStr == 'Q':
            print("Exiting...")
            break


        try:
            inStrAsSeparate = inStr.split(" ") # split by spaces
            if len(inStrAsSeparate) != 3:
                print("There is NOT exactly two numbers with an operator in the middle, and the three are separated by spaces.")
                continue
            # Set the three separate strings into num1, operator, and num2 respectively
            num1, operator, num2 = inStrAsSeparate
            num1 = float(num1)
            num2 = float(num2)
        except IndexError:
            print("Something went wrong with your expression")
            continue
        except ValueError:
            print("Numbers are not floating point numbers. Please enter NUMBERS!")
            continue

        # Use operator to choose the function
        if operator == '+':
            answer = add(num1, num2)
        elif operator == '-':
            answer = subtract(num1, num2)
        elif operator == '-':
            answer = subtract(num1, num2)
        elif operator == '-':
            answer = subtract(num1, num2)
        else:
            print("You did not type an OPERATOR (+, -. *, /)! Please type one of those operators")
            continue

        print("Result: " answer)

File: Lab1/1001761950_calculator.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

//Prototypes for the functions add, multiply, subtract, and divide
double add(double num1, double num2);
double multiply(double num1, double num2);
double subtract(double num1, double num2);
double divide(double num1, double num2);

double getNum();
char getOp();


int main()
{
    double num1, num2, answer = 0;
    char operator, choice;
    int cont = 1; //use for the do-while loop

    do
    {
        //ask user for the numbers and the operators
        printf("Enter a mathematical expression (i.e. 1 + 1): ");
        if(scanf("%lf %c %lf", &num1, &operator, &num2) != 3) //make sure there is at least three
        {
            printf("WRONG! Type a MATH expression (2 numbers separated by an operator).\n");
            continue; //if they didn't write an expression using exactly three elements -> redo the loop
        }
        
        while ( getchar() != '\n' ); // clear scanf buffer. Putting this here as check cause it keeps not reading the scanf for choice later on.

        //perform the calculator depending on the operator
        switch (operator)
        {
            case '+':
                answer = add(num1, num2);
                break;
            case '-':
                answer = subtract(num1, num2);
                break;
            case '*':
                answer = multiply(num1, num2);
                break;
            case '/':
                if(num2 == 0)
                {
                    printf("You can't divide by 0 in this universe!\n");
                    continue; //break out of the matrix
                }
                answer = divide(num1, num2);
                break;
            default:
                printf("Operator is INVALID");
                break;
        }

        //print answer
        printf("%.2lf %c %.2lf = %.2lf\n", num1, operator, num2, answer);

        //ask if the user wants to do another calculation
        printf("Would you like to calculate another expression? (Y/n): ");
        scanf("%c", &choice);
        if(choice == 'n' || choice == 'N')
        {
            cont = 0;
            break;
        }else
        {
            printf("\n");
        }
    } while (1);

    printf("Exiting...\n");
    
    
    

    return 0;
}

double add(double num1, double num2)
{
    return num1 + num2;
}

double subtract(double num1, double num2)
{
    return num1 - num2;
}

double multiply(double num1, double num2)
{
    return num1 * num2;
}

double divide(double num1, double num2)
{
    if(num2 == 0)
    {
        return 0;
    }
    return num1 / num2;
}